spec/lib/catalog/submit_order_item_spec.rb

```diff
@@ -101,4 +101,82 @@
       expect { submit_order.process }.to raise_exception(Catalog::InvalidSurvey)
     end
   end
+
+  context "with multiple order items" do
+    let(:params) { order.id.to_s }
+    let(:order_response) { TopologicalInventoryApiClient::InlineResponse200.new(:task_id => "200") }
+    let(:order_items) do
+      (1..3).map do |sequence|
+        create(:order_item_with_callback,
+               :order            => order,
+               :service_plan_ref => service_plan_ref,
+               :process_sequence => sequence,
+               :portfolio_item   => portfolio_item)
+      end
+    end
+
+    before do
+      stub_request(:post, topological_url("service_offerings/998/order"))
+        .to_return(:status => 200, :body => order_response.to_json, :headers => {"Content-type" => "application/json"})
+
+      allow(Order).to receive(:find_by!).with(:id => params).and_return(order)
+      allow(order).to receive(:order_items).and_return(order_items)
+    end
+
+    shared_examples_for "order the desired item" do
+      it "orders the desired item" do
+        submit_order.process.order.order_items.each.with_index(1) do |item, index|
+          if index == ordered_item
+            expect(item.state).to eq('Ordered')
+          else
+            expect(item.state).not_to eq('Ordered')
+          end
+        end
+      end
+    end
+
+    context "when no order item is orderable" do
+      before do
```

Comment:
How about move `before` block into `shared_example` , based on `index`, we already know which one is `false` or `true`?